pymc/model.py

```diff
@@ -1303,14 +1303,15 @@ def register_rv(
         else:
             if (
                 isinstance(data, Variable)
-                and not isinstance(data, (GenTensorVariable, Minibatch))
+                and not isinstance(data, (GenTensorVariable))
                 and data.owner is not None
                 # The only Aesara operation we allow on observed data is type casting
                 # Although we could allow for any graph that does not depend on other RVs
                 and not (
                     isinstance(data.owner.op, Elemwise)
                     and isinstance(data.owner.op.scalar_op, Cast)
                 )
+                and not data.name.startswith("minibatch")
```

Comment:
You're not using a custom TensorType anymore? If so you can check against that.

Reply:
no custom tensor type(

Reply:
I think `_joint_logp` will complain when it sees the RVs in the logp graph? Have you not seen that?

Reply:
If you see that, the easiest would be to subclass the Uniform RV used in the Minibatch Slice and allow that type here:

https://github.com/pymc-devs/pymc/blob/5da26173428cacc6aa5e1d05e4b51ceb69dadfd9/pymc/distributions/logprob.py#L107-L126

Reply:
I've added tests that only pass after I implement minibatching with a new RV